### PR TITLE
[docs]Improve document of non-persistent topic

### PR DESCRIPTION
**Pull request by tuteng**
### Motivation

Improve the documentation for Non-Persistent Topic

### Modifications

* Improve error codes and error messages
* Description Information of Improvement Parameters

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API: (yes / no)
  - The schema: (yes / no / don't know)
  - The default values of configurations: (yes / no)
  - The wire protocol: (yes / no)
  - The rest endpoints: (yes / no)
  - The admin cli options: (yes / no)
  - Anything that affects deployment: (yes / no / don't know)

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

**Comment by Jennifer88huang** [comment]
Sure. @Anonymitaet could you please help review this file? Thank you.

**Comment by Anonymitaet** [comment]
@sijie @Anonymitaet Glad to help.

**Comment by tuteng** [comment]
run java8 tests

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/NonPersistentTopics.java [hunk 1]
```diff
@@ -66,9 +67,22 @@
     @GET
     @Path("/{tenant}/{namespace}/{topic}/partitions")
     @ApiOperation(value = "Get partitioned topic metadata.")
-    @ApiResponses(value = { @ApiResponse(code = 403, message = "Don't have admin permission") })
-    public PartitionedTopicMetadata getPartitionedMetadata(@PathParam("tenant") String tenant,
-            @PathParam("namespace") String namespace, @PathParam("topic") @Encoded String encodedTopic,
+    @ApiResponses(value = {
+            @ApiResponse(code = 401, message = "Don't have permission to manage resources on this tenant"),
+            @ApiResponse(code = 403, message = "Don't have admin permission"),
+            @ApiResponse(code = 404, message = "tenant/namespace/topic does not exist"),
```
Comment (Anonymitaet) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/NonPersistentTopics.java]:
```suggestion
            @ApiResponse(code = 404, message = "Tenant/namespace/topic does not exist"),
```

Please check all similar cases.

**Comment by tuteng** [comment]
Thank you very much for your review.  @Anonymitaet PTAL

**Pull request merged**